packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -66,6 +67,18 @@ enum RoutePopDisposition {
   bubble,
 }
 
+/// Name for the method which is used for sending messages from framework to
```

Comment:
It seems like in the web platform, 'routePopped' and 'routePushed' do the same thing, setting the current route in the web platform. Why not just have a single method, eg 'updateRoute'? Are you planning on adding more navigation features in web later which will use the different methods and the newRoute/previousRoute?

Reply:
Thanks for the suggestion.

The web platform are not using them identically:
switch (decoded.method) {
          case 'routePopped':
            webOnlyRouteName = previousRouteName;
            return;
          case 'routePushed':
            webOnlyRouteName = newRouteName;
            return;
        }

will be the code. Replace and push is similar but we thought with (@mdebbar) the way we will handle these two operations can change in the future too.